api/controllers/v2/github.ts

```diff
@@ -79,6 +133,133 @@ export const TriggerBuild = async (event: APIGatewayEvent): Promise<APIGatewayPr
+    };
+  }
+
+  if (body.workflow_run.conclusion != 'success')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body: `Build on branch ${body.workflow_run.head_branch} is not complete and will not trigger smoke test site deployments`,
+    };
+
+  if (body.workflow_run.name != 'Deploy Staging ECS')
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body: `Workflow ${body.workflow_run.name} completed successfully but only Deploy Staging ECS workflow completion will trigger smoke test site deployments`,
+    };
+
+  // if the build was not building main branch, no need for smoke test sites
+  if (body.workflow_run.head_branch != 'main' || body.repository.fork) {
+    console.log('Build was not on master branch in main repo, sites will not deploy as no smoke tests are needed');
+    return {
+      statusCode: 202,
+      headers: { 'Content-Type': 'text/plain' },
+      body: `Build on branch ${body.workflow_run.head_branch} will not trigger site deployments as it was not on main branch in upstream repo`,
+    };
+  }
+
+  //automated test builds will always deploy in dotcomstg
+  const env = 'dotcomstg';
+
+  async function createAndInsertJob() {
+    return await Promise.all(
+      SMOKETEST_SITES.map(async (repoName): Promise<string> => {
+        const jobTitle = 'Smoke Test ' + repoName;
+        let repoInfo, projectEntry, repoOwner;
+        try {
+          repoInfo = await docsetsRepository.getRepo(repoName);
+          projectEntry = await projectsRepository.getProjectEntry(repoInfo.project);
+          repoOwner = projectEntry.github.organization;
+        } catch (err) {
+          consoleLogger.error(
+            `Atlas Repo Information Error`,
+            `RepoInfo, projectEntry, or repoOwner not found for docs site ${repoName}. RepoInfo: ${repoInfo}, projectEntry: ${projectEntry}, repoOwner: ${repoOwner}`
+          );
+          return err;
+        }
+
+        const jobPrefix = repoInfo?.prefix ? repoInfo['prefix'][env] : '';
+        const payload = await createPayload({
+          repoName,
+          isSmokeTestDeploy: true,
+          prefix: jobPrefix,
+          repoBranchesRepository,
+          repoInfo,
+          repoOwner,
+        });
+
+        //add logic for getting master branch, latest stable branch
+        const job = await prepGithubPushPayload(body, payload, jobTitle);
+
+        try {
+          consoleLogger.info(job.title, 'Creating Job');
+          const jobId = await jobRepository.insertJob(job, c.get('jobsQueueUrl'));
+          jobRepository.notify(jobId, c.get('jobUpdatesQueueUrl'), JobStatus.inQueue, 0);
+          consoleLogger.info(job.title, `Created Job ${jobId}`);
+          return jobId;
+        } catch (err) {
+          consoleLogger.error('TriggerBuildError', `${err} Error inserting job for ${repoName}`);
+          return err;
+        }
+      })
+    );
+  }
+
+  let returnVal;
```

Comment:
NIT: We don't need to use `let returnVal` here. We can return in the `try` statement with the `returnVal`